packages/rules-engine/src/lib/custom-rule-engine/index.ts

```diff
@@ -0,0 +1,29 @@
+import { CustomRuleSet } from '@/lib/custom-rule-engine/types/custom-rule-set';
+import { uboHasWarning } from '@/lib/custom-rule-engine/operations/ubo-has-warning';
+import { CustomRuleInput } from '@/lib/custom-rule-engine/types/custom-rule-input';
+import { evaluate } from './operations/evaluate';
+
+export const CustomRuleEngine = ({
+  context,
+  ruleset,
+}: {
+  context: Record<string, unknown>;
+  ruleset: CustomRuleSet;
+}) => {
+  const operationMethod = ruleset.operator === 'or' ? 'some' : 'every';
+
+  return ruleset.rules
+    .map(rule => {
+      return invokeRule(rule, context);
+    })
+    [operationMethod](Boolean);
```

Comment:
This might make it a little more readable IMO:

```suggestion
  const invokedRules = ruleset.rules
    .map(rule => {
      return invokeRule(rule, context);
    });
    
  if (ruleset.operator === 'or') {
    return invokedRules.some(Boolean);
  }
  
  return invokedRules.every(Boolean);
```

Reply:
@Blokh Remember `safeEvery`.

Reply:
@Omri-Levy Why `safeEvery`?